### PR TITLE
Make result name suggestion for new msg waits

### PR DESCRIPTION
**Pull request by ericnewcomer**
Fixes: #156 (this still needs unit tests)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "allowJs": true,
     "watch": true,
     "alwaysStrict": true,
-    "lib": ["es6", "dom"]
+    "lib": ["es2017", "dom"]
```
Comment (ycleptkellan) [inline tsconfig.json]: why the change here?

Comment (ericnewcomer) [reply tsconfig.json]: `Object.values()` .. so certainly not necessary, but don't see any reason to avoid it. 

More here: https://github.com/Microsoft/TypeScript/issues/10768

Comment (ericnewcomer) [reply tsconfig.json]: Hmmm.. but that is failing the node 6 build, so maybe there's one.

Comment (ycleptkellan) [reply tsconfig.json]: nice, can't wait for that to land

Comment (ycleptkellan) [reply tsconfig.json]: `Object.entries()`?

File: src/component/Flow.tsx [hunk 2]
```diff
@@ -330,9 +340,30 @@ export default class Flow extends React.Component<FlowProps, FlowState> {
 
             ghost.actions.push(replyAction);
         } else {
+            // coun't up all our msg waits
+            let responseNumber = 1;
```
Comment (ycleptkellan) [inline src/component/Flow.tsx]: should these be broken-out into helper methods, e.g. `getResponseNumber` and `getResultName`? would that make testing easier?

File: src/component/NodeEditor/NodeEditor.tsx [hunk 3]
```diff
@@ -764,11 +765,9 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
     }
 
     public open(): void {
-        const config = getTypeConfig(this.determineConfigType(this.props));
-
         this.setState({
-            show: true,
-            config
+            ...this.getInitialState(),
```
Comment (ycleptkellan) [inline src/component/NodeEditor/NodeEditor.tsx]: yes eric, yes...embrace the spread operator...

Comment (ericnewcomer) [reply src/component/NodeEditor/NodeEditor.tsx]: special case!

**Comment by ericnewcomer** [comment]
@ycleptkellan what say we merge this so we don't have anything outstanding against master

**Comment by ycleptkellan** [comment]
aye, merge away

File: src/component/Flow.tsx [hunk 4]
```diff
@@ -294,6 +304,30 @@ export default class Flow extends React.Component<FlowProps, FlowState> {
         this.resetState();
     }
 
+    /**
+     * Gets a suggested result name based on the current number of waits
+     * in the current definition
+     */
+    public getSuggestedResultName(): string {
```
Comment (ycleptkellan) [inline src/component/Flow.tsx]: @ericnewcomer what think you about–in the future–shying away from adding methods to component classes that don't either handle a callback that changes state somewhere in the render tree or determine whether or not JSX is rendered? my reasoning is that a method that doesn't read from the component's `this` context is much easier to test (no need to render `Flow`, access it's instance and then test `getSuggestedResultName`). what do you think?

File: src/component/Flow.test.tsx [hunk 5]
```diff
@@ -40,5 +40,16 @@ describe('Flow >', () => {
             expect(getSpecWrapper(wrapper, 'nodes').hasClass('node_list')).toBeTruthy();
             expect(wrapper.find('NodeComp')).toBeTruthy();
         });
+
+        it('should suggest appropriate response names', () => {
+            const wrapper = shallow(<Flow {...props} />, {
```
Comment (ycleptkellan) [inline src/component/Flow.test.tsx]: Alt Text

File: src/component/Flow.tsx (hunk 4 above)
Comment (ericnewcomer) [reply src/component/Flow.tsx]: Agreed.
